src/wp-includes/rest-api/endpoints/class-wp-rest-application-passwords-controller.php

```diff
@@ -373,6 +389,70 @@ public function delete_item( $request ) {
 		);
 	}
 
+	/**
+	 * Checks if a given request has access to get the currently used application password.
+	 *
+	 * @since 5.7.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access for the item, WP_Error object otherwise.
+	 */
+	public function get_current_item_permissions_check( $request ) {
+		$user = $this->get_user( $request );
+
+		if ( is_wp_error( $user ) ) {
+			return $user;
+		}
+
+		if ( get_current_user_id() !== $user->ID ) {
+			return new WP_Error(
+				'rest_cannot_introspect_app_password_for_non_authenticated_user',
+				__( 'The authenticated Application Password can only be introspected for the current user.' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Retrieves the application password being currently used for authentication.
```

Comment:
I'm trying to work the purpose of this function out, shouldn't the app know the password already by its ability to reach this point?

Reply:
The app knows its password, but not its `uuid`. The `uuid` is how they are queried, but the password is how they are authenticated.